tooling/image-updater/update.sh

```diff
@@ -0,0 +1,272 @@
+#!/bin/bash
+set -euo pipefail
+
+CREATE_PR=true
+COMPONENTS=""
+EXCLUDE_COMPONENTS=""
+
+if [ -n "${GITHUB_SERVER_URL:-}" ] && [ -n "${GITHUB_REPOSITORY:-}" ] && [ -n "${GITHUB_RUN_ID:-}" ]; then
+    WORKFLOW_URL="${GITHUB_SERVER_URL}/${GITHUB_REPOSITORY}/actions/runs/${GITHUB_RUN_ID}"
+    AUTOMATION_CREDIT="PR created automatically with [Image Digest Updater](${WORKFLOW_URL})"
+else
+    AUTOMATION_CREDIT="Automatically updated with Image Digest Updater"
+fi
+
+log() {
+    echo "[$(date +'%H:%M:%S')] $1" >&2
+}
+
+usage() {
+    cat << EOF
+Usage: $0 [OPTIONS]
+
+Bulk update all component image digests and create a single PR.
+
+OPTIONS:
+    --components NAMES          Update only specified components (comma-separated)
+    --exclude-components NAMES  Exclude specified components from update (comma-separated)
+    --no-pr                     Skip PR creation (useful for local testing)
+    -h, --help                  Show this help message
+
+EXAMPLES:
+    $0                                              # Update all components and create PR
+    $0 --no-pr                                     # Update all components but don't create PR
+    $0 --components arohcpfrontend,arohcpbackend   # Update only frontend and backend
+    $0 --exclude-components arohcpfrontend,arohcpbackend  # Update all except frontend and backend
+EOF
+}
+
+parse_args() {
+    while [[ $# -gt 0 ]]; do
+        case $1 in
+            --components)
+                COMPONENTS="$2"
+                shift 2
+                ;;
+            --exclude-components)
+                EXCLUDE_COMPONENTS="$2"
+                shift 2
+                ;;
+            --no-pr)
+                CREATE_PR=false
+                shift
+                ;;
+            -h|--help)
+                usage
+                exit 0
+                ;;
+            *)
+                log "❌ Unknown option: $1"
+                usage
+                exit 1
+                ;;
+        esac
+    done
+}
+
+get_current_date() {
+    date '+%Y-%m-%d'
+}
+
+cleanup_previous_prs() {
+    log "🔍 Checking for previous image update PRs to close..."
+
+    local existing_prs
+    existing_prs=$(gh pr list --state open --search "updated image components in:title" --json number,title --jq '.[].number' 2>/dev/null || echo "")
+
+    if [[ -n "$existing_prs" ]]; then
+        log "📝 Found existing image update PRs, closing them..."
```

Comment:
Please update them in-place instead of closing

Reply:
Ask @deads2k  and the DPTP team for where the source code is for managing declaratively some set of PRs that is used by OCP automation - you can import that library and use it, it's got years of battle behind it. 10000x preferrable to shell.